homepage/homepage/content/docs/using-covalues/imagedef/react.mdx

```diff
@@ -78,194 +82,229 @@ The `createImage()` function:
+  resolve: {
+    original: true,
+  },
 });
 
-// Accessing the highest available resolution
-const highestRes = ImageDefinition.highestResAvailable(image);
-if (highestRes) {
-  console.log(`Found resolution: ${highestRes.res}`);
-  console.log(`Stream: ${highestRes.stream}`);
-}
+console.log({
+  originalSize: image.originalSize,
+  placeholderDataUrl: image.placeholderDataUrl,
+  original: image.original, // this FileStream may be not loaded yet
+});
 ```
 </CodeGroup>
 
-For more details on using `ImageDefinition` directly, see the [VanillaJS docs](/docs/vanilla/using-covalues/imagedef).
+`image.original` is a `FileStream` and its content can be read as described in the [FileStream](/docs/react/using-covalues/filestreams#reading-from-filestreams) documentation.
 
-### Fallback Behavior
+Since FileStream objects are also CoValues, they must be loaded before use. To simplify loading, if you want to load the binary data saved as Original, you can use the `loadImage` function.
 
-`highestResAvailable` returns the largest resolution that fits your constraints. If a resolution has incomplete data, it falls back to the next available lower resolution.
+<CodeGroup>
+```ts
+import { loadImage } from "jazz-tools/media";
+
+const image = await loadImage(imageDefinitionOrId);
+
+console.log({
+  width: image.width,
+  height: image.height,
+  image: image.image,
+});
+```
+</CodeGroup>
+
+If the image was generated with progressive loading, and you want to access the best resolution, `loadImageBySize` function is here the utility function you need. It will load the image of the best resolution that fits the wanted width and height.
 
 <CodeGroup>
-```ts twoslash
-import { ImageDefinition, FileStream } from "jazz-tools";
-const mediumSizeBlob = new Blob([], { type: "image/jpeg" });
+```ts
+import { loadImageBySize } from "jazz-tools/media";
 
+const image = await loadImageBySize(imageDefinitionOrId, 600, 600);
+
+console.log({
+  width: image.width,
+  height: image.height,
+  image: image.image,
+});
+```
+</CodeGroup>
+
+
+But you know, the CoValue are alive and they will change over time. If want to listen to the changes and retrieve the best resolution _loaded_ that fits the wanted width and height, you can use the `subscribe` and the `highestResAvailable` function.
+
+<CodeGroup>
+```ts
+import { ImageDefinition } from "jazz-tools";
 // ---cut---
-const image = ImageDefinition.create({
-  originalSize: [1920, 1080],
+// function highestResAvailable(image: ImageDefinition, wantedWidth: number, wantedHeight: number): FileStream | null
+import { highestResAvailable } from "jazz-tools/media";
+
+const image = await ImageDefinition.load("123");
+
+image.subscribe({}, (image) => {
+  const bestImage = highestResAvailable(image, 600, 600);
+  if(bestImage) {
+    // bestImage is again a FileStream
+    const blob = bestImage.toBlob();
+    if(blob) {
+      const url = URL.createObjectURL(blob);
+      // ...
+    }
+  }
 });
+```
+</CodeGroup>
+
+
```

Comment:
Maybe add a section here about how to create "resized *copies*" of an image?

something like `createImage(await existing.toBlob(), { maxSize: 500 })` (pseudo-code)